### ISSUE TITLE
[Question]: SD WebUI plugin -- Error Message: connect ECONNREFUSED

### ISSUE THREAD

**Issue by NoahDragon** [danny-avila/LibreChat#493]
### What is your question?

The plugins demo for 0.5.0 is amazing, and can't wait to try it.

When I used the SD WebUI plugin, but it seems not work, when I checked the log and found out an error message about ECONNREFUSED:
```
Error Message: connect ECONNREFUSED 0.0.0.0:7860
```

Wondering if there is any sepcial config need to be set in docker compose yaml file?

### More Details

The SD webui has run with `--api --listen` arguments and hosted in host 0.0.0.0:7860.

I have used the docker compose and pulled the latest images from dockerhub. For the `SD_WEBUI_URL` in `.env` file, I have tried the following values, all got the same error:
* http://0.0.0.0:7860
* http://localhost:7860
* http://host.docker.internal:7860

I also tried to set the api service in docker compose file to use `network_mode: "host"`, but this will cause the meilisearch error.

### What is the main subject of your question?

Installation

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by danny-avila** [comment]
Hey thanks for trying out the project! 

`SD_WEBUI_URL=http://host.docker.internal:7860` did in fact work for me. My docker-compose file is set to the default configuration, with nothing changed.

You're right that it shouldn't work with localhost or 0.0.0.0 from inside docker, and I need to update the documentation for this. 

The issue you're facing is a common one when working with Docker. The IP address `127.0.0.1` or `localhost` inside a Docker container refers to the container itself, not the host machine. So when your application inside the Docker container tries to connect to `127.0.0.1:7860`, it's trying to connect to itself, not your host machine where the API is running.

Usually to resolve: 

1. Use `host.docker.internal` - worked for me but this didn't work for you for some reason, do you have stable diffusion containerized?

2. **Use the host machine's IP address**: You can find this by running `ipconfig` (Windows) or `ifconfig` (Unix). Replace `127.0.0.1` with this IP in your application.

`127.0.0.1` or `localhost` inside a Docker container refers to the container itself, not the host machine. So, your application in the Docker container is trying to connect to itself, not your host machine where the API is running.

**Comment by danny-avila** [comment]
You may also try uncommenting these lines from the docker-compose file (or commenting if they are potentially creating the issue)

```yaml
 extra_hosts: # if you are running APIs on docker you need access to, you will need to uncomment this line and next
 - "host.docker.internal:host-gateway"
```

According to ChatGPT:

The `extra_hosts` directive in a Docker configuration allows you to add entries to the `/etc/hosts` file in your Docker containers. This can be used to provide custom host-to-IP mappings.

The line `- "host.docker.internal:host-gateway"` is adding a mapping that resolves `host.docker.internal` to the IP address of the host machine's network gateway. This is essentially another way to allow a Docker container to communicate with the host machine.

This is particularly useful if you're running Docker in a Linux environment, where the `host.docker.internal` hostname doesn't work out of the box like it does on Docker for Windows and Docker for Mac.

By uncommenting these lines, you're telling Docker to add an entry to the `/etc/hosts` file in your container that maps `host.docker.internal` to the IP address of the host machine's network gateway. This means that from within your Docker container, you can use `http://host.docker.internal:7860` to access your API running on the host machine.

However, please note that the `host-gateway` value for `extra_hosts` is only available from Docker 20.10 onwards. If you're using an older version of Docker, you might need to manually specify the IP address of the host machine.

**Comment by NoahDragon** [comment]
Awesome, it works!

I have done the following:
* Using `SD_WEBUI_URL=http://host.docker.internal:7860` in the `.env` file
* Uncomment the following in `docker-compose.yml` file
```
 extra_hosts: # if you are running APIs on docker you need access to, you will need to uncomment this line and next
 - "host.docker.internal:host-gateway"
```

Thank you so much for the quick response, and detailed explaination!
